src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/FactsMapper.java

```diff
@@ -84,8 +84,26 @@ public String mapUpdateSessionToRuleJsonMessage(Session session) throws JsonProc
         return objectMapper.writeValueAsString(factSession);
     }
 
-    public String mapHearingToRuleJsonMessage(Hearing hearing) throws JsonProcessingException {
-        return mapDbHearingToRuleJsonMessage(hearing);
+    public String mapHearingToRuleJsonMessage(HearingPart hearingPart) throws JsonProcessingException {
+        FactHearingPart factHearingPart = new FactHearingPart();
+        Hearing hearing = hearingPart.getHearing();
+
+        factHearingPart.setId(hearingPart.getId().toString());
+        factHearingPart.setDuration(hearing.getDuration());
+        factHearingPart.setCaseTypeCode(hearing.getCaseType().getCode());
+        factHearingPart.setHearingTypeCode(hearing.getHearingType().getCode());
+        factHearingPart.setScheduleStart(hearing.getScheduleStart());
+        factHearingPart.setScheduleEnd(hearing.getScheduleEnd());
+        factHearingPart.setCreatedAt(hearing.getCreatedAt());
+
+        if (hearingPart.getSessionId() != null) {
+            factHearingPart.setSessionId(hearingPart.getSessionId().toString());
+        }
+
+        Optional.ofNullable(hearingPart.getSession()).ifPresent(
```

Comment:
isn;t this a duplicate of the code above?

Reply:
I know thta by default it would be null, but I would rather prefer to actually set it to null so it;s obvious.